### PR TITLE
per task disable or lenient verification

### LINKED ISSUE #3940: Add options to task creation dictionary to make the requestor more lenient towards the verification results
* add an `x-disable-verification` option to the task creation parameters - when specified and `True` - it should completely disable verification for the given task and treat all results as valid without performing any verification - default `False`
* add an `x-lenient-verification` option to the task creation parameters - when specified, the verification should still be performed and then in the event of negative verification result - the subtask itself should be marked as failed as usual and rescheduled and the provider should also be banned from performing any additional subtasks in this task and a failure should still be reported to the golem monitor - BUT - when this flag is present in the task definition - the provider should get a `SubtaskResultsAccepted` response and be issued a payment just as it would had the result been correct. In other words - this flag should cause the verification result to seem positive from the provider's POV.

Those options should only be known to the requestor itself. The options should NOT be exposed in the `TaskHeader` and should be clearly marked in the accompanying comments within the code as experimental.

### PR DESCRIPTION
**Pull request by etam**
Resolves #3940 

Blocked by #3630 

Testing plan: I think the best way to test this is to make proper integration tests. For this I need a verificator that always fails (I don't think that we could have a task that always and reliably fails verification) and an option to force using it.

### CODE REVIEW

**Comment by etam** [comment]
Comment by @shadeofblue: for integration test, instead of failing verificator, we could have a provider that always returns bogus files (for example provided 1 black pixel).

File: apps/core/task/coretaskstate.py [hunk 1]
```diff
@@ -13,6 +13,28 @@
 DEFAULT_SUBTASK_TIMEOUT = 20 * 60
 
 
+class RunVerification(enum.Enum):
+    """
+    Disabled:
```
Comment (shadeofblue) [inline apps/core/task/coretaskstate.py]: @etam I'd also add a docstring for `Enabled` and say it's a default to make it perfectly clear what's going on here

Comment (etam) [reply apps/core/task/coretaskstate.py]: done

File: apps/core/task/coretask.py [hunk 2]
```diff
@@ -208,10 +210,19 @@ def computation_finished(self, subtask_id, task_result,
...
 
-        def verification_finished_(subtask_id, verdict, result):
+        def verification_finished_(subtask_id,
+                                   verdict: SubtaskVerificationState, result):
             self.verification_finished(subtask_id, verdict, result)
             verification_finished()
 
+        if self.task_definition.run_verification == RunVerification.disabled:
+            logger.debug("subtask %s: verification disabled; "
+                         "calling verification_finished", subtask_id)
```
Comment (jiivan) [inline apps/core/task/coretask.py]: It's only debug, but I would follow the convention from https://github.com/golemfactory/golem-internal/wiki/Logging-guidelines#log-message-structure

File: golem/appconfig.py [hunk 3]
```diff
@@ -201,6 +201,8 @@ def load_config(cls, datadir, cfg_file_name=CONFIG_FILENAME):
             hyperdrive_address=DEFAULT_HYPERDRIVE_ADDRESS,
             hyperdrive_rpc_port=DEFAULT_HYPERDRIVE_RPC_PORT,
             hyperdrive_rpc_address=DEFAULT_HYPERDRIVE_RPC_ADDRESS,
+            # testing
```
Comment (jiivan) [inline golem/appconfig.py:204]: ?

**Review by jiivan — APPROVED** [review]
Looks good :+1:

File: golem/appconfig.py (hunk 3 above)
Comment (etam) [reply golem/appconfig.py:204]: What's the problem?

Comment (jiivan) [reply golem/appconfig.py:204]: I don't understand this comment. Could you ELIF? :)

Comment (etam) [reply golem/appconfig.py:204]: You can see above, there are comments, that are some sort of section headers.

File: apps/core/task/coretask.py (hunk 2 above)
Comment (etam) [reply apps/core/task/coretask.py]: 1) What exactly you have in mind? Which point?
2) Why is this guideline in private repo?

Comment (jiivan) [reply apps/core/task/coretask.py]: ad1.
> Always start with a message, print data as "kwargs". For example Task deadline passed. id=%r, state=%r

ad2. dunno

Comment (etam) [reply apps/core/task/coretask.py]: done

**Comment by etam** [comment]
I rebased changes onto latest develop and squashed some commits.

**Pull request merged**